sonar-db/src/test/java/org/sonar/db/component/ComponentTesting.java

```diff
@@ -29,13 +30,13 @@
 
 public class ComponentTesting {
 
-  public static ComponentDto newFileDto(ComponentDto subProjectOrProject) {
-    return newFileDto(subProjectOrProject, Uuids.create());
+  public static ComponentDto newFileDto(ComponentDto subProjectOrProject, @Nullable ComponentDto directory) {
```

Comment:
To create a valid File component, we need to know the first parent module (which can be the project) and the first parent (which can be the module/project). Both informations are required and keeping only the methods which emphasis this will avoid creating File component with some incorrect fields.